src/form/Input/InlineInput/InlineInput.tsx

```diff
@@ -40,22 +51,54 @@ export interface InlineInputProps
 export const InlineInput: FC<InlineInputProps> = forwardRef(
   (
     {
+      className,
       inputClassName,
-      placeholderIsMinWidth = true,
+      placeholder,
+      placeholderIsMinWidth,
       theme: customTheme,
+      extraWidth,
+      onChange,
       ...rest
     },
-    ref: Ref<HTMLInputElement>
+    ref: RefObject<InputRef>
   ) => {
     const theme: InputTheme = useComponentTheme('input', customTheme);
+    const spanStyles = 'relative opacity-0 whitespace-pre select-none';
+    const [inputValue, setInputValue] = useState('');
+    const [placeholderValue, setPlaceholderValue] = useState(placeholder);
+
+    const onInputValueChange = event => {
+      setInputValue(event.target.value);
+      onChange(event);
+    };
+
+    const setRef = (node: HTMLInputElement) => {
+      ref?.current?.inputRef;
+
+      const placeholder = node?.placeholder;
+      const value = node?.value;
+
+      placeholder && setPlaceholderValue(placeholder);
+      value && setInputValue(value);
+    };
 
     return (
-      <AutosizeInput
-        inputRef={ref}
-        inputClassName={twMerge(theme.inline, inputClassName)}
-        placeholderIsMinWidth={placeholderIsMinWidth}
-        {...rest}
-      />
+      <div className={twMerge(className, theme.inline.base)}>
+        <span className={spanStyles} style={{ paddingRight: extraWidth ?? 0 }}>
+          {inputValue}
+        </span>
+        {!placeholderIsMinWidth === true && !inputValue && (
+          <span className={spanStyles}>{placeholder}</span>
+        )}
```

Comment:
This `!placeholderIsMinWidth === true` seems backwards. Now a component like this has a width of 0 and the placeholder isn't visible

```
InlineInput
  placeholder="Type here..."
  placeholderIsMinWidth
/>
```